feat(#1798): community section, first time contrib and related

mrjones-plip opened the pull request:
<!--
Please use semantic PR titles that respect this format:

<type>(#<issue number>): <subject>

Quick example:

feat(#1234): add hat wobble
^--^(#^--^): ^------------^
|     |      |
|     |      + - > subject
|     |
|     + -------- > issue number
|
+ -------------- > type: chore, feat, fix, perf.

https://docs.communityhealthtoolkit.org/contribute/code/workflow/#commit-message-format
-->

# Description

This PR:
* Adds a new top level `/community` section 
* Creates a new `/community/first-time-contributors` page
* moves the [existing hall of fame](https://docs.communityhealthtoolkit.org/contribute/code/hall-of-fame/) under this new section


per #1798

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

andrablaj commented on content/en/contribute/code/_index.md:
```diff
@@ -52,7 +44,7 @@ Want to talk about Documentation generally? Join our [Community Forum](https://f
 ### Translations
 
-If you are a translator but not a developer, we understand that you may need extra help to follow the [process of translating]({{% ref "building/translations/managing" %}}) software for the first time. If that is the case, please open an issue on the GitHub repo or start a topic on the community forum.
+If you are a translator but not a developer, we understand that you may need extra help to follow the [process of translatingsoftware for the first time. If that is the case, please open an issue on the GitHub repo or start a topic on the community forum.
```
It looks like the link to the translations page is broken. Please review it.

mrjones-plip replied:
good catch - thank you!  fixed in fdfe0b5f

mrjones-plip commented:
oh no! did I add you as a reviewer when I created this @andrablaj? sorry! It's not ready yet :sweat_smile: 

happy to get your early feedback though - thank you! 

stay tuned - when I'm ready I'll ask for another review

andrablaj commented:
No, you didn't add me as a reviewer; I saw the change and sneak-peeked at what this looked like (as I am super excited about this new `Community` section 😍 ). Then I saw the typo, and I figured I would let you know. Take your time, ignore me and add as reviewer the people you had planned to. 😄

andrablaj commented:
@mrjones-plip Flagging the ["Contributing" page](https://docs.communityhealthtoolkit.org/contribute/) that I think should be moved under `Community`. 

Similarly, all the contributing code/doc "home" pages should be merged:
- https://docs.communityhealthtoolkit.org/contribute/code/
- https://docs.communityhealthtoolkit.org/contribute/docs/

What do you think?

mrjones-plip commented:
oh yeah - how did I miss all the "contributor" content when I [pulled in](https://github.com/medic/cht-docs/pull/1818/commits/fffd98285d03f6ce76509506e7fa6e1c6785f54e#diff-018b276e61a188becf9651d2cd7951a0b4037ebb6578a5ef1aa9bdad22bb353d) that "first time contrib" content :thinking: . Thanks for pointing it out Andra!

Yeah...this should all be in one spot.  I'll see what I can do.  Thanks again!

Phil-Mwago commented:
@mrjones-plip  Here is content on CHT community journeys. 
Tried to make it simple, harmonized and easy to follow based on our chat the other day.
Where will we place it? New page? 

# CHT Community Journeys

## Community Member Journey  
**Objective:** Guide community members from discovery and beginers to thriving community contributors.  

| Phase               | Key Activities                                                                 | Resources Involved                          |
|----------------------|-------------------------------------------------------------------------------|--------------------------------------------|
| **Initial Interaction** | - Promote via CHT website, newsletters, social media.                         | Technical partners                 |
|                      | - Communicate community value (e.g., influence on roadmap).                   |                              Forum, website and social media.              |
| **Onboarding**       | - Welcome message + demo site access.                                         | Onboarding team, Technical partners, Volunteers.                           |
|                      | - Monthly onboarding calls.                                                   |          Sign up form                                  |
| **Engagement**       | - **Forums**: Q&A, feature requests.                                          | Community dashboard                                                     |
|                      | - Participatory round up call, Participatory technical webinar, Share industry news, Diverse forum categories, Monthly email with call to action, Connect to opportunities, Promote continuous learning on academy. | Community calls/platforms
| **Sustained Engagement** | - Community Meetups, webinars, squad calls roundup calls.                                  | Technical partners                         |
|                      | - CHT Academy courses & early feature access.                                 |
|                      | - **Feedback loops**: Surveys, recognition for contributors.                  |
| **Thriving Contributor** | - Fellowship programs & governance roles.             | Governance team, Ambassadors, Contributors.                            |
|                      | - Accreditization, solution partnerships.                              |                                            |

---

## Squad Journey  
**Objective:** The community comes together to collaboratively build new features or enhance existing features.

| Phase             | Key Activities                                                                 | Resources Involved                          |
|-------------------|-------------------------------------------------------------------------------|--------------------------------------------|
| **Inception**     | Feedback collection, Identify common needs within the community, Analysis on emerging requirements from the community, Research on whether this requirements have been raised before and what was the outcome, Reaching out to potential community members/organizations on importance of these requirements, Find out willingness to join and contribute to the community, Add items to a roadmap                               | Feedback platforms                         |                                          |
| **Convergence**   | -Have the identified TP or community members for the initial convergence meeting, Draft a scope of work that includes the envisioned workflow based on requirements, Assign roles, Define the timelines, Create a tracking board for the activities, Create a github repository                         | Technical partners                         |
| **Building**      | - Weekly dev sprints; progress tracking. Active communication to ensure there are no blockers.                                      | Devs, QA                                   |
| **Testing**       | - QA/UAT; verify against criteria.                                            | Technical partners                                |
| **Release**       | - Deploy to production; advocate via blogs/forums,Verification to add to subsequent releases CHT releases.|Technical partners 
| **Measure/Sunset**| - Monitor impact/usage; sunset features if needed.                                   |      N/A                                      |

---

### **Connecting Both Journeys**  
- **Feedback Flow**: Member requests → Squad inception.  
- **Advocacy**: Squad releases → Member early access.  
- **Governance**: Thriving contributors join squads and Governance team.

mrjones-plip commented:
@andrablaj and @Phil-Mwago - thanks thanks!  I'm still completing content and need to add `aliases` for all the redirects, but please check my latest commit [1cca6fc](https://github.com/medic/cht-docs/pull/1818/commits/1cca6fc138bdb2288877ddbe855c18d89d35aa31) and give a shout if you see anything of big concern!  I'll otherwise press ahead to finish this up by tomorrow EOD (Fri 11th).

What my latest commit [1cca6fc](https://github.com/medic/cht-docs/pull/1818/commits/1cca6fc138bdb2288877ddbe855c18d89d35aa31)  does in words:
* moves `contribute/code` -> `community/code` 
* moves `contribute/docs` -> `community/docs` 
* adds `community/code/first-time-contributors` 
* updates `community` to have Andra's [questions](https://github.com/medic/cht-docs/pull/1818#issuecomment-2794327486) and appends Phil's [journey content](https://github.com/medic/cht-docs/pull/1818#issuecomment-2795294200)


What my latest commit [1cca6fc](https://github.com/medic/cht-docs/pull/1818/commits/1cca6fc138bdb2288877ddbe855c18d89d35aa31)  in pictures:

![image](https://github.com/user-attachments/assets/d27268c2-9a0b-486d-ac18-711fbabbda0c)

![image](https://github.com/user-attachments/assets/829138f8-c9fe-4408-b5b6-0e442bb57bda)

andrablaj commented:
Thanks for the updates, @mrjones-plip! This already looks very good. I appreciate your efforts in making the docs site more welcoming! ❇️ 

Some comments:

- I suggest moving the "/community/" content to a new page called "Welcome to the community" or "Get started" to make it obvious that that's the page where someone can find out how to get involved. 
- I suggest moving `Code` & `Documentation` & `Technical Resources` sections under a new sub-section called `Contributing` as they are a bit misleading to sit just under `Community`. 

The proposed hierarchy is:
- Community
  - Get started
  - Contributing
    - Code of Conduct
    - Technical Resources
    - Code
    - Documentation
  - Tech radar 
  - Hall of Fame

Additionally, I suggest we discuss the `Journey` additions from @Phil-Mwago separately (in another PR), as I am unsure if that content & language will be helpful for the CHT Community. I believe the community would benefit from knowing how to form a squad and how to contribute to the squad. For example, I think we should move the content from the [CHT Roadmap](https://github.com/medic/cht-roadmap) repo to the docs site as that language is more actionable and easier to understand.

In consequence, please remove the Journey sections from this PR and let's continue that discussion separately.

mrjones-plip commented:
Awesome - thanks so much Andra!  Really appreciate the detailed and helpful feedback.  Stay tuned!

mrjones-plip commented:
I've refactored the structure but am breaking out "[Disclosing vulnerabilities](https://docs.communityhealthtoolkit.org/contribute/code/#disclosing-vulnerabilities)" to it's own page and merging most of the "[ways to contribute](https://docs.communityhealthtoolkit.org/contribute/code/#ways-to-contribute)" into the `_index.md` under "Contributing".  Having a separate "How to Contribute" page on top of an index page seems excessive.

Feeling a bit warmer with this PR finally!  Added a "Quick Start" section, followed by a "First time Contributors Guide".  I had a slow day today, so I'll pick up some hours this weekend to have it all ready by Monday!

![image](https://github.com/user-attachments/assets/f9b3fc80-cd1f-4046-b9af-12b62799d8e5)

mrjones-plip commented:
@andrablaj and @Phil-Mwago - I'd love some feedback on the new content!  

Andra - you'll be interested just about everything.  I still need to add the `alias` lines so all the redirects work, but the content should be good to go.

Phil - you'll primarily be interested in the [community landing page](https://github.com/medic/cht-docs/blob/0e3769340ca2abcc74b0b69f38232626db2be2af/content/en/community/_index.md), but I welcome feedback on everything!

andrablaj commented on content/en/community/contributing/_index.md:
```diff
@@ -0,0 +1,214 @@
+
+1. Create a Fork
+
+    Fork this repository by clicking on the fork button on the top of this page.
```
It's unclear to me what "this repository" means. Is this cht-core? or any CHT repo?

mrjones-plip replied:
yes - good point!  changed to:

> Create your own fork the CHT Core repository by going to the CHT's [fork page on GitHub](https://github.com/medic/cht-core/fork). When you click "Create Fork" on that page, you will  create a copy of this repository in your GitHub account.

and below updated: 

> where "url you just copied" (without the quotation marks) is the url to the repository (your fork of the CHT Core). See the previous steps to obtain the url.

see 5d90eec and 73b2659

andrablaj commented on content/en/community/contributing/code/core/_index.md:
```diff
@@ -1,5 +1,5 @@
 ---
-title: "Contributing CHT Core Code"
+title: "contributing CHT Core Code"
```
```suggestion
title: "Contributing CHT Core Code"
```

andrablaj commented on content/en/community/contributing/disclosing-vulnerabilities.md:
```diff
@@ -0,0 +1,47 @@
+---
+
+
+We take the security of our systems seriously, and we value the security community. The disclosure of security vulnerabilities helps us ensure the security and privacy of our users.
```
```suggestion
In the CHT Community, we take the security of our systems seriously and value the security community. Disclosure of security vulnerabilities helps us ensure the security and privacy of the CHT users.
```

andrablaj replied:
suggestion: Make it more straightforward who "we" is.

andrablaj commented on content/en/community/contributing/disclosing-vulnerabilities.md:
```diff
@@ -0,0 +1,47 @@
+---
```
Great call to make this a separate page!

mrjones-plip replied:
thanks!

andrablaj reviewed (changes requested):
Thank you @mrjones-plip for the thorough work; this will make it so much easier for newbies to navigate the CHT! 

I added a few suggestions, mainly around separating some pages. Let me know what you think. 🎇

mrjones-plip commented on content/en/community/_index.md:
```diff
@@ -0,0 +1,44 @@
+---
```
Breaking this  content into it's own page would leave the `/community` page empty (and especially empty with the move to [Hextra](https://github.com/medic/cht-docs/pull/1830)).   With this content letting you easily self identify who you are, having as the entry way and the first thing you see, keeping it on "Community" makes sense to me. 

Gently push back on this recommendation!

unresolve and comment if you still disagree!

mrjones-plip commented on content/en/community/contributing/_index.md:
```diff
@@ -0,0 +1,214 @@
+---
+title: "Contributing code and documentation"
```
since we have the quick start guide, agreed!

see 5d90eec

mrjones-plip commented:
Merged `main` so there should be no merge conflicts and content should be up to date when comparing it to live site. also added all the aliases so the `check.urls.sh` is green.  tended all the feedback - back to you @andrablaj !

andrablaj commented on content/en/community/_index.md:
```diff
@@ -0,0 +1,46 @@
+---
+title: "Welcome to the community"
```
I love this page and I think we can reformat it nicely with [Hextra cards](https://imfing.github.io/hextra/docs/guide/shortcodes/) when time comes!

andrablaj commented on content/en/community/contributing/code/android/_index.md:
```diff
@@ -1,9 +1,11 @@
 ---
-title: "Contributing Android Code"
+title: "contributing Android Code"
```
```suggestion
title: "Contributing Android Code"
```

andrablaj commented on content/en/community/contributing/code/core/_index.md:
```diff
@@ -4,4 +4,7 @@ linkTitle: "CHT Core"
 weight: 3
 description: >
   How to contribute to code to the CHT Core Framework
+
```
```suggestion
```

andrablaj commented on content/en/community/contributing/code/releasing/feature_releases.md:
```diff
@@ -5,10 +5,9 @@ weight: 1
 relatedContent: >
-    releases/
-    contribute#are-you-a-partner-wondering-how-issues-are-prioritized
+       releases/
```
```suggestion
    releases/
```

andrablaj reviewed (approved):
Good to go! 🚀 

I added a few nitpicks, nothing blocking!

mrjones-plip commented:
Thanks for all the feedback Andra! much appreciated
